docs/src/modules/components/Demo.js

```diff
@@ -263,150 +265,161 @@ function Demo(props) {
 
   const match = useMediaQuery(theme => theme.breakpoints.up('sm'));
 
+  /* regex matches the content of the return statement in the default export
+   * `export default.*`
+   * `return (\n` or `return `
+   *  everything (not greedy), until:
+   * `  );
+   * }`
+   * or
+   * `;
+   * }`
+   */
+  let jsxOnly = demoData.raw
+    ? demoData.raw.match(/export default .*(return \(\n|return )(.*?)(\n {2}\);\n}|;\n})/s)
+    : null;
+  jsxOnly = jsxOnly ? jsxOnly[2] : '';
+  const codeLength = jsxOnly.split(/\n/).length;
+  const codeShort = jsxOnly !== '' && codeLength <= 20;
+
   return (
     <div className={classes.root}>
+      <div
+        className={clsx(classes.demo, {
+          [classes.demoHiddenHeader]: demoOptions.hideHeader,
+        })}
+        tabIndex={-1}
+        onMouseEnter={handleDemoHover}
+        onMouseLeave={handleDemoHover}
+      >
+        <DemoSandboxed
+          style={demoSandboxedStyle}
+          component={DemoComponent}
+          iframe={demoOptions.iframe}
+          name={demoName}
+        />
+      </div>
       <div className={classes.anchorLink} id={`${demoName}.js`} />
       <div className={classes.anchorLink} id={`${demoName}.tsx`} />
       {demoOptions.hideHeader ? null : (
-        <div>
-          <div className={classes.header}>
-            <DemoLanguages
-              demo={demo}
-              codeOpen={codeOpen}
-              codeVariant={codeVariant}
-              gaEventCategory={gaCategory}
-              onLanguageClick={handleCodeLanguageClick}
-            />
-            <div>
-              <Tooltip
-                classes={{ popper: classes.tooltip }}
-                key={showSourceHint}
-                open={showSourceHint && match ? true : undefined}
-                PopperProps={{ disablePortal: true }}
-                title={codeOpen ? t('hideSource') : t('showSource')}
-                placement="top"
+        <div className={classes.header}>
+          <DemoLanguages
+            demo={demo}
+            codeOpen={codeOpen}
+            codeVariant={codeVariant}
+            gaEventCategory={gaCategory}
+            onLanguageClick={handleCodeLanguageClick}
+          />
+          <div className={classes.headerButtons}>
+            <Tooltip
+              classes={{ popper: classes.tooltip }}
+              key={showSourceHint}
+              open={showSourceHint && match ? true : undefined}
+              PopperProps={{ disablePortal: true }}
+              title={codeOpen ? t('hideSource') : t('showSource')}
```

Comment:
When a small example is shown "Show Source" doesn't make much sense. Should this been "Show Full Source" when codeShort is true?

Reply:
Yep. This was confusing for me as well.